openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -59,6 +60,62 @@ const Carousel = {
             slidesToScroll: a,
             responsive: responsive_settings
         });
+
+        var availabilityStatuses = {
+            'open': {cls: 'cta-btn--available', cta: 'Read'},
+            'borrow_available': {cls: 'cta-btn--available', cta: 'Borrow'},
+            'borrow_unavailable': {cls: 'cta-btn--unavailable', cta: 'Join Waitlist'},
+            'error': {cls: 'cta-btn--missing', cta: 'No eBook'}
+        };
+
+        var addWork = function(work) {
+            var availability = work.availability.status;
+            var ocaid = work.availability.identifier;
+            var cls = availabilityStatuses[availability].cls;
+            var url = (cls == 'cta-btn--available') ?
+                ('/borrow/ia/' + ocaid) : (cls == 'cta-btn--unavailable') ?
+                    ('/books/' + work.availability.openlibrary_edition) : work.key;
```

Comment:
SUGGESTION: Multiple ternary operators is a code smell. Consider more explicit `if-else`'s or `switch`.